service/src/test/java/bio/terra/catalog/rawls/RawlsServiceTest.java

```diff
@@ -47,4 +66,59 @@ void statusException() throws Exception {
     var rawlsStatus = rawlsService.status();
     assertFalse(rawlsStatus.isOk());
   }
+
+  @Test
+  void getWorkspaces() throws Exception {
+    var items = Map.of("id", List.of("role"));
+    var workspaceResponses =
+        List.of(
+            new WorkspaceListResponse()
+                .workspace(new WorkspaceDetails().workspaceId("id"))
+                .accessLevel(WorkspaceAccessLevel.OWNER));
+    when(workspacesApi.listWorkspaces(List.of("accessLevel", "workspace.workspaceId")))
+        .thenReturn(workspaceResponses);
+    assertThat(rawlsService.getWorkspaceIdsAndRoles(user), is(workspaceResponses));
+  }
+
+  @Test
+  void getSnapshotsException() throws Exception {
+    when(workspacesApi.listWorkspaces(List.of("accessLevel", "workspace.workspaceId")))
+        .thenThrow(new ApiException());
+    assertThrows(RawlsException.class, () -> rawlsService.getWorkspaceIdsAndRoles(user));
+  }
+
+  @Test
+  void userHasActionReader() throws Exception {
+    var id = UUID.randomUUID();
```

Comment:
```suggestion
    var stringId = UUID.randomUUID().toString();
```

Possible suggestion, I think it simplifies visually scanning these tests and makes L119 more straightforward.